### DISCUSSION TITLE
ECDH-ES (X25519) supported?

### DISCUSSION THREAD

**Discussion by awoie** [panva/jose#55] [Q&A]
Does the jose package has support for ECDH-ES (X25519)? In the documentation you mention you have implemented RFC8037.

**Comment by panva** [comment]
The readme explicitly says CFRG Elliptic Curve Signatures (EdDSA) from RFC8037 are implemented

The detailed matrix says

> Pending Node.js Support 🤞:
> 
> - ECDH-ES with X25519 and X448 - see nodejs/node#26626

When Node.js supports it, its going to be supported by jose too. Please go to that issue and voice your interest in the feature.

**Comment by panva** [comment]
@awoie i've refactored and with ^1.16.0 there's the option to install a plugin that adds x25519 ECDH-ES support.

https://github.com/panva/jose#plugins
https://github.com/panva/jose-x25519-ecdh

also one for the chacha individual draft

https://github.com/panva/jose-chacha

**Comment by awoie** [comment]
That is very helpful! We will try it out!

**Comment by panva** [comment]
`^1.23.0` now supports ECDH-ES for X25519 and X448 without any plugins when Node.js runtime with the new and improved `crypto.diffieHellman` interface is detected.
